Modify Nuget spec file to meet the new compliance validation check

xuzhg opened the pull request:
<!-- markdownlint-disable MD002 MD041 -->

### Issues

* Modify Nuget spec file to meet the new compliance validation check

### Description

*Briefly describe the changes of this pull request.*

1 | The author must be just   “Microsoft” without quotes. | <authors>Microsoft</authors>
-- | -- | --
2 | The copyright must be “©   Microsoft Corporation. All rights reserved.” without quotes. | <copyright>©   Microsoft Corporation. All rights reserved.</copyright>
3 | projectUrl must not be null | <projectUrl>This   must be a valid team or product URL. This must be a permalink that will never   be deprecated. It can be an FWLink.</projectUrl>
4 | licenseUrl must not be null. | <licenseUrl>This   must be a valid product license URL. While FWLinks are not prohibited, customers   prefer for these URLs to be transparent rather than being FWLinks.   </licenseUrl>
5 | the nupkg must be signed with an   official Microsoft certificate | [Learn more](https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki?wikiVersion=GBwikiMaster&pagePath=%2FProduct%20Development%20Reference%2FNuGet&anchor=2.-create-an-organization-account-at-%5Bnuget.org%5D)
6 | reserve the necessary ID   prefixes for your account | [Learn more](https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki?wikiVersion=GBwikiMaster&pagePath=%2FProduct%20Development%20Reference%2FNuGet&anchor=2.-create-an-organization-account-at-%5Bnuget.org%5D)
7 | the package must be owned by a   team org account and not by individuals | [Learn more](https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki?wikiVersion=GBwikiMaster&pagePath=%2FProduct%20Development%20Reference%2FNuGet&anchor=2.-create-an-organization-account-at-%5Bnuget.org%5D)


### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

xuzhg commented:
@mikepizzo @biaol-odata @raheph @KanishManuja-MS @madansr7 Would you please take a quick look about this PR.

biaol-odata commented on src/Microsoft.OData.Client/Build.NuGet/Microsoft.OData.Client.Nightly.Release.nuspec:
```diff
@@ -5,16 +5,15 @@
     <version>$VersionFullSemantic$-Nightly$NightlyBuildVersion$</version>
-    <authors>Microsoft Corporation</authors>
-    <owners>Microsoft Corporation</owners>
+    <authors>Microsoft</authors>
```
why "Microsoft Corporation" -> "Microsoft"?

biaol-odata commented:
To provide proper context, please add the following url as the "Learn more" link.
https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki?wikiVersion=GBwikiMaster&pagePath=%2FProduct%20Development%20Reference%2FNuGet&anchor=2.-create-an-organization-account-at-%5Bnuget.org%5D

biaol-odata commented:
Looks like the changes are item #1, #2 in the table above, and removing <owners> properties per the "Learn more" link. No changes related to other items in the description table. Correct?

xuzhg commented:
Yes. changed for item 1, item 2, removed the "owners" , because it says "Do not populate this field in the package nuspec file. ". It looks we don't change anything for item 3, 4, 5, 6, 7.

xuzhg commented:
Added the link for "Learn more", Thanks.
